src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -24,19 +24,21 @@
  */
 package org.spongepowered.api.plugin;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.regex.Pattern;
 
 /**
  * An annotation used to describe and mark a Sponge plugin.
  */
-@Target(TYPE)
-@Retention(RUNTIME)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
 public @interface Plugin {
 
+    Pattern ID_PATTERN = Pattern.compile("[a-z0-9-_]+");
```

Comment:
It would work I guess, but people will want to change their packages in some cases, while the plugin ID is short and would only really change if the plugin name changes.

I personally doubt we would be hitting conflicts here any time soon but if we really want to prevent them we could just do it similar to build systems like Maven and require an additional group to be specified similar like normal Java packages, e.g.:

```java
@Plugin(group = "net.minecrell", id = "testplugin", dependencies = @Dependency(group = "com.sk89q", id = "worldedit"))
```